logstash-core/src/main/java/org/logstash/config/ir/PipelineIR.java

```diff
@@ -63,6 +66,9 @@ public PipelineIR(Graph inputSection, Graph filterSection, Graph outputSection,
         // Now we connect the queue to the root of the filter section
         tempGraph = tempGraph.chain(filterSection);
 
+        // Connect the filter section to the filter end vertex to separate from the output section
+        tempGraph = tempGraph.chain(new SeparatorVertex("filter_to_output"));
```

Comment:
if we are hard-coding the id of all separator vertices to a single constant-value string, does it need to be a required parameter to the constructor method? ~And if `SeparatorVertex` is only ever instantiated with a constant `id` and carries no other state, can we reliably use a static singleton instead of instantiating a new one for each pipeline?~ (EDIT: superclass `Vertex` has much state)

Reply:
As I commented above, my idea was that a SeparatorVertex could possibly used for multiple purposes, and even replace the QueueVertex, so the ID becomes the differentiator. I originally had a specific "FilterToOutputVertex" but figured these "separators" (that one and the QueueVertex) had the same purpose in the end so I opted to make it multi purpose. Maybe that's not the best approach and a more OO strategy with something like 
```java
abstract class SeparatorVertex extends Vertex {}

class FilterToOutputVertex extends SeparatorVertex{}

//and eventually
class QueueVertex extends SeparatorVertex {}
```
WDYT?

Reply:
If the 2 subclasses (`FilterToOutputVertex` and `QueueVertex`) are only placeholders and don't provide any behaviour probably is best to introduce an enum with the flavour of these vertices, this is more friendly in if/switch than `instanceof`, at least we are sure we don't put behaviour in those 2 subclasses

Reply:
good idea ... but `enum` can't extend `Vertex` so we would beed to refactor this even more, unless I am missing something simpler here?  If not I suggest we keep it like this and defer the `Vertex` / `SeparatorVertex` to another PR ? WDYT?

Reply:
I meant to add a `SeparatorType getType()` method to `SeparatorVertex` and define the `enum SeparatorType QUEUE, FILTER_TO_OUTPOUT`. But probably is best to make this happen in another PR, else we complicate this too much